music_rnn/src/model.ts

```diff
@@ -191,8 +198,9 @@ export class MusicRNN {
         samples.push(nextInput.as1D().toBool());
       }
 
-      if (controlSignal) {
-        const control = controlSignal.getTensor(i + 1);
+      if (controls) {
+        const control =
+            controls.slice([i, 0], [1, controlSize]).as2D(1, controlSize);
```

Comment:
It should be more efficient if we just return a list for the controls instead of a 2D tensor.

Reply:
Talked offline, leaving 2D tensor for now but may change in multiple places later.